sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GeneratePredicate.scala

```diff
@@ -55,9 +55,15 @@ object GeneratePredicate extends CodeGenerator[Expression, (InternalRow) => Bool
           ${initMutableStates(ctx)}
         }
 
-        public boolean eval(InternalRow ${ctx.INPUT_ROW}) {
+        public PredResult eval(InternalRow ${ctx.INPUT_ROW}) {
```

Comment:
Sorry for the late reply.
For anti join, we need to take care of the predicte, as we need to filter out the `null` value, however, `null` will be cast into `false` if we said the return type of `eval` is `boolean`.